### PR TITLE
Implement AP startup code

### PR DESCRIPTION
**Pull request by Extravert-ir**
## Purpose
This PR is a continuation of #3795

## TODO

- [x] The second AP starts properly, calling `KiSystemStartup`
- [ ] Implement basic debugger support for SMP (`KeFreezeExecution`, `KeThawExecution` & friends)
- [ ] Check stuff works on GCC builds
- [ ] Parse ACPI tables to get information about CPUs
- [ ] Implement apentry.S for amd64
- [ ] More code cleanup

### CODE REVIEW

File: sdk/include/ndk/kefuncs.h [hunk 1]
```diff
@@ -73,6 +73,13 @@ KiDeliverApc(
 //
 // Process/Thread Functions
 //
+CODE_SEG("INIT")
+VOID
+NTAPI
+KeStartAllProcessors(
```
Comment (HBelusca) [inline sdk/include/ndk/kefuncs.h:79]: This is an function internal to ntoskrnl and is not an export --> it should not be placed in this file, but instead in ntoskrnl/includes/internal/ke.h

File: ntoskrnl/ntos.cmake [hunk 2]
```diff
@@ -323,6 +323,10 @@ if(ARCH STREQUAL "i386")
         ${REACTOS_SOURCE_DIR}/ntoskrnl/ps/i386/psldt.c
         ${REACTOS_SOURCE_DIR}/ntoskrnl/vdm/vdmmain.c
         ${REACTOS_SOURCE_DIR}/ntoskrnl/vdm/vdmexec.c)
+    if(BUILD_MP)
```
Comment (HBelusca) [inline ntoskrnl/ntos.cmake:326]:
```suggestion
    if(CONFIG_SMP)
```

File: ntoskrnl/ke/i386/mproc.c [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        Entry->HighWord.Bits.Granularity = 1;
+    }
+    Entry->LimitLow = (UINT16)(Limit & 0xffff);
+    Entry->HighWord.Bits.LimitHi = ((Limit >> 16) & 0x0f);
+}
+
+CODE_SEG("INIT")
+VOID
+NTAPI
+KeStartAllProcessors()
```
Comment (HBelusca) [inline ntoskrnl/ke/i386/mproc.c:75]:
```suggestion
KeStartAllProcessors(VOID)
```

File: ntoskrnl/ke/i386/mproc.c [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    DECLSPEC_ALIGN(PAGE_SIZE) KIDTENTRY Idt[256];
+    KTSS Tss;
+    KTSS TssDoubleFault;
+    KTSS TssNMI;
+    DECLSPEC_ALIGN(16) UINT8 NMIStackData[DOUBLE_FAULT_STACK_SIZE];
+} APINFO, *PAPINFO;
+
+/* GLOBALS *******************************************************************/
+
+extern PLOADER_PARAMETER_BLOCK KeLoaderBlock;
```
Comment (HBelusca) [inline ntoskrnl/ke/i386/mproc.c:29]: It's already available ntoskrnl-wide without this declaration.

File: hal/halx86/apic/apicsmp.c [hunk 5]
```diff
@@ -1,15 +1,16 @@
 /*
  * PROJECT:     ReactOS HAL
  * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
- * FILE:        hal/halx86/apic/apicsmp.c
  * PURPOSE:     SMP specific APIC code
- * PROGRAMMERS: Copyright 2021 Timo Kreuzer (timo.kreuzer@reactos.org)
+ * COPYRIGHT:   Copyright 2021 Timo Kreuzer (timo.kreuzer@reactos.org)
+ *              Copyright 2021 Justin Miller (justinmiller100@gmail.com)
```
Comment (binarymaster) [inline hal/halx86/apic/apicsmp.c:6]:
```suggestion
 * COPYRIGHT:   Copyright 2021 Timo Kreuzer <timo.kreuzer@reactos.org>
 *              Copyright 2021 Justin Miller <justinmiller100@gmail.com>
```

File: hal/halx86/generic/up.c [hunk 6]
```diff
@@ -0,0 +1,32 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Core source file for UP alternative functions
+ * COPYRIGHT:  Copyright 2021 Justin Miller <justinmiller100@gmail.com>
```
Comment (binarymaster) [inline hal/halx86/generic/up.c:5]:
```suggestion
 * COPYRIGHT:   Copyright 2021 Justin Miller <justinmiller100@gmail.com>
```

**Pull request closed without merging**